app/assets/javascripts/reducers/stories.js

```diff
@@ -74,14 +74,26 @@ const storiesReducer = (state = initialState, action) => {
             addNewAttributes(story, { ...action.story, needsToSave: false, loading: false })
           ))
       })
+    case actionTypes.SORT_STORIES_SUCCESS:
+      return allScopes(state, null, stories => {
+        return stories.map((story) => {
+          for(let i = 0; i < action.stories.length; i+=1) {
+            const incomingStory = action.stories[i];
+            if (incomingStory.id === story.id) {
+              return addNewAttributes(story, { ...incomingStory, needsToSave: false, loading: false })
+            }
+          }
+          return story;
```

Comment:
Looks like we could use `find` here instead of a `for` :thinking:

Reply:
`find` will only return the first element that satisfies the passed condition. In this case, we are dealing with one or many elements to be changed

Reply:
We can use `filter` then, right?

Reply:
Actually, for this case, `find` looks like it's the correct to use. You use the for to find _one_ element where the if matches what you're looking for (hence the fact you return after finding it instead of keep going).

Reply:
Got it